commit 3b2990810bdb5742787b73cd7c7cbde67cbaf59a (refs/changes/60/360/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2016-08-26T15:39:01-07:00 (3 years, 1 month ago)
    
    DOCKER-920 Docker build shouldn't need to use chown after copying files

diff --git a/bin/docker-build.js b/bin/docker-build.js
index 6b7555e..de45682 100644
--- a/bin/docker-build.js
+++ b/bin/docker-build.js
@@ -273,6 +273,12 @@ function runBuild(opts) {
 
     log.info('%s - started for zone %s', commandType, opts.uuid);
 
+    // Before we start, we want all files created by docker build/commit to use
+    // 'root:root' permissions. Note that the cn-agent (parent) process defaults
+    // to using 'root:staff' permissions.
+    process.setuid(0);
+    process.setgid(0);
+
     sendMessage('Starting ' + commandType, {socket: socket, log: log});
 
     if (commandType === 'commit') {
@@ -745,7 +751,7 @@ function handleExtractTarfileEvent(builder, event) {
     var tarfile = event.tarfile;
 
     // Ensure the holding (parent) directory exists.
-    builder.mkdirpChown(extractDir, function (err) {
+    mkdirp(extractDir, function (err) {
         if (err) {
             callback(err);
             return;
